core/src/main/scala/neotypes/Driver.scala

```diff
@@ -18,35 +18,44 @@ import scala.jdk.CollectionConverters._
   * @tparam F effect type for driver
   */
 sealed trait Driver[F[_]] {
+  def transactionConfig: TransactionConfig
+
   def metrics: F[List[ConnectionPoolMetrics]]
 
   def transaction(config: TransactionConfig): F[Transaction[F]]
 
-  final def transaction: F[Transaction[F]]  =
-    transaction(config = TransactionConfig.default)
+  final def transaction: F[Transaction[F]] =
+    transaction(config = transactionConfig)
 
   def transact[T](config: TransactionConfig)(txF: Transaction[F] => F[T]): F[T]
 
   final def transact[T](txF: Transaction[F] => F[T]): F[T] =
-    transact(config = TransactionConfig.default)(txF)
+    transact(config = transactionConfig)(txF)
 
   /** Close the resources assigned to the neo4j driver.
     *
     *  @return an effect F of Unit.
     */
   def close: F[Unit]
+
+  def withTransactionConfig(config: TransactionConfig): Driver[F]
 }
 
 sealed trait StreamingDriver[S[_], F[_]] extends Driver[F] {
   def streamingTransaction(config: TransactionConfig): S[StreamingTransaction[S, F]]
 
   final def streamingTransaction: S[StreamingTransaction[S, F]] =
-    streamingTransaction(config = TransactionConfig.default)
+    streamingTransaction(config = transactionConfig)
 
   def streamingTransact[T](config: TransactionConfig)(txF: StreamingTransaction[S, F] => S[T]): S[T]
 
   final def streamingTransact[T](txF: StreamingTransaction[S, F] => S[T]): S[T] =
-    streamingTransact(config = TransactionConfig.default)(txF)
+    streamingTransact(config = transactionConfig)(txF)
+
+  /** Creates a new driver instance using the provided transaction configuration as default. NB. This does NOT create
+   * a new instance of the underlying Java driver.
+   */
+  def withTransactionConfig(config: TransactionConfig): StreamingDriver[S, F]
```

Comment:
I believe we could _(should?)_ move this **svaladoc** comment to the method defined in `Driver` and inherit the documentation here.
Also, IIRC, we can mark this method as `override` and still leave it abstract.